### PR TITLE
Fix clippy errors for Rust 1.80

### PR DESCRIPTION
**Pull request by findepi**
Newest version of clippy complains when list item continuation is not indented.

Fixes https://github.com/apache/datafusion/issues/11651

Closes https://github.com/apache/datafusion/pull/11467

### CODE REVIEW

**Comment by findepi** [comment]
Cherry picked https://github.com/apache/datafusion/pull/11467 and rebased it on current main.

File: datafusion/sqllogictest/test_files/parquet.slt [hunk 1]
```diff
@@ -251,25 +251,26 @@ SELECT COUNT(*) FROM timestamp_with_tz;
 ----
 131072
 
+# FIXME(#TODO) fails with feature `force_hash_collisions`
```
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/sqllogictest/test_files/parquet.slt:254]:
```suggestion
# FIXME(#TODO) fails with feature `force_hash_collisions`
# https://github.com/apache/datafusion/issues/11660
```

**Review by alamb — APPROVED** [review]
Thank you @findepi  (and @nrc )

This looks good to me. 

I filed follow on tickets, and I'll make a subsequent PR to annotate the tests with them so it is easier to understand why they are commented out (without resorting to git archeology)

File: datafusion/physical-plan/src/joins/hash_join.rs [hunk 2]
```diff
@@ -2097,6 +2102,8 @@ mod tests {
     }
 
     /// Test where the left has 1 part, the right has 2 parts => 2 parts
+    // FIXME(#TODO) test fails with feature `force_hash_collisions`
```
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/physical-plan/src/joins/hash_join.rs:2105]: in in https://github.com/apache/datafusion/pull/11661

**Pull request merged**